### PR TITLE
fix: adjust `/* #__PURE__*/` comment timing

### PR DESCRIPTION
**Pull request by newives**

### CODE REVIEW

File: packages/core-base/src/context.ts [hunk 1]
```diff
@@ -414,14 +414,11 @@ export function registerLocaleFallbacker(fallbacker: LocaleFallbacker): void {
 // Additional Meta for Intlify DevTools
 let _additionalMeta: MetaInfo | null = /* #__PURE__*/ null
 
-export const setAdditionalMeta = /* #__PURE__*/ (
-  meta: MetaInfo | null
-): void => {
+export const setAdditionalMeta = (meta: MetaInfo | null): void => {
```
Comment (kazupon) [inline packages/core-base/src/context.ts:417]: I think we swap to `/* #__NO_SIDE_EFFECTS__ */`, not remove `/* #__PURE__ */`
see the https://github.com/intlify/vue-i18n-next/issues/1599#issuecomment-1773776151
ref: https://github.com/javascript-compiler-hints/compiler-notations-spec/blob/main/no-side-effects-notation-spec.md

**Review by kazupon — CHANGES_REQUESTED** [review]

File: packages/core-base/src/compilation.ts [hunk 2]
```diff
@@ -52,7 +52,7 @@ function baseCompile(
   return { ...baseCompileCore(message, options), detectError }
 }
 
-export const compileToFunction = /* #__PURE__*/ <
+export const compileToFunction = <
```
Comment (kazupon) [inline packages/core-base/src/compilation.ts:55]: same https://github.com/intlify/vue-i18n-next/pull/1623#pullrequestreview-1707899783

File: packages/core-base/src/context.ts [hunk 3]
```diff
@@ -414,14 +414,11 @@ export function registerLocaleFallbacker(fallbacker: LocaleFallbacker): void {
...
   _additionalMeta = meta
 }
 
-export const getAdditionalMeta = /* #__PURE__*/ (): MetaInfo | null =>
-  _additionalMeta
+export const getAdditionalMeta = (): MetaInfo | null => _additionalMeta
```
Comment (kazupon) [inline packages/core-base/src/context.ts:421]: same https://github.com/intlify/vue-i18n-next/pull/1623#pullrequestreview-1707899783

**Review by kazupon — CHANGES_REQUESTED** [review]

**Comment by newives** [comment]
swap to `/* #__NO_SIDE_EFFECTS__ */`, not remove /* #__PURE__ */ Local test packaging will report the following error

图片

**Comment by newives** [comment]
done

**Review by kazupon — APPROVED** [review]
Well done!
Thanks!

**Pull request merged**